developer_docs/order_docs.md

```diff
@@ -0,0 +1,427 @@
+Both of the orders discussed here (`order_f` and `order_c`) are stored **contiguously** in memory, but they differ in how they are arranged.
+
+### Order F
+
+`order='F'` tells `Numpy` to store the array column by column (column-major), example:  
+
+```python
+import numpy as np
+
+if __name__ == "__main__":
+  a = np.array([[1, 2, 3],
+                [4, 5, 6],
+                [7, 8, 9]], order='F')
+  print(a.ravel('K'))
+```  
+
+`array.ravel('k')` shows us how the array is stored in memroy, this python script will output `[1 4 7 2 5 8 3 6 9]`, notice that the columns are stored one after the other.  
+
+### Order C
+
+`order='C'` on the other hand tells `Numpy` to store the array row by row (row-major), example:  
+
+```python
+import numpy as np
+
+if __name__ == "__main__":
+  a = np.array([[1, 2, 3],
+                [4, 5, 6],
+                [7, 8, 9]], order='C') # order='C' is the default in numpy.array
+  print(a.ravel('K'))
+ ```  
+ 
+This python script will output `[1 2 3 4 5 6 7 8 9]`, notice that the rows are stored one after the other.
+
+### Printing and indexing in `Numpy`
+
+`order` in `Numpy` does not affect the indexing or the printing, unlike `transposing`, the `shape` of the array remains the same, only the `strides` change, example:  
+
+```python
+import numpy as np
+
+if __name__ == "__main__":
+   a = np.array([[1, 2],
+                [4, 5],
+                [7, 8]], order='F')
+   b = np.array([[1, 2],
+                [4, 5],
+                [7, 8]], order='C')
+   print(a.shape, a.strides) # output: (3, 2) (8, 24)
+   print(b.shape, b.strides) # output: (3, 2) (16, 8)
+   print(a)
+   # output:[[1 2]
+   #         [4 5]
+   #         [7 8]] 
+   print(b)
+   # output:[[1 2]
+   #         [4 5]
+   #         [7 8]]
+   
+   print(a[2][1], a[0][0], a[1]) # output: 8 1 [4 5]
+   print(b[2][1], b[0][0], b[1]) # output: 8 1 [4 5]
+```  
+
+`arr.strides` is the variable that helps us navigate the array (indexing/printing), it tells us how many bytes we have to skip in memory to move to the next position along a certain axis (dimension). For example for `memory_layout_of_a = [1 4 7 2 5 8]` and `strides_of_a = (8, 24)`, we have to skip 8 bytes (1 element for `int64`) to move to the next row, but 24 bytes (3 elements for `int64`) to get to the same position in the next column of `a`.  
+`a[2][1]` would give us `'8'`, using the `strides`: `2 * 8 + 1 * 24 = 40`, which means that in the flattened array, we would have to skip `40` bytes to get the value of `a[2][1]`, each element is 8 bytes, so we would have to skip `40 / 8 = 5` elements to get to `'8'`
+
+## Pyccel's C code
+
+In `Pyccel`'s C code, we try to clone `Numpy`'s indexing/printing and memory layout conventions.
+
+### Ordering in C code
+
+The arrays in `C` code, are flattened into a one dimensional string, `strides` and `shape` are used to navigate the array (unlike `Numpy`. `Pyccel`'s strides use 'number of elements' instead of 'number of bytes' as a unit)
```

Comment:
A string? I don't think so. What object are you talking about here?

Reply:
By `string` I meant a `series of something` since that's one of its definitions, but I can see how that can be confusing in programming docs!
We could say `array` instead

Reply:
1d array? Flattened array?

Reply:
Yeh, we change it from `are flattened into a one dimensional string` to ` are flattened into a one dimensional arry`